docs/api/apiv3/components/examples/membership-schema-response.yml

```diff
@@ -0,0 +1,67 @@
+# Example: Membership schema response
+---
+value:
+  _type: 'Schema'
+  _dependencies: [ ]
+  id:
+    type: 'Integer'
+    name: 'ID'
+    required: true
+    hasDefault: false
+    writable: false
+    options: { }
+  createdAt:
+    type: 'DateTime'
+    name: 'Created on'
+    required: true
+    hasDefault: false
+    writable: false
+    options: { }
+  updatedAt:
+    type: 'DateTime'
+    name: 'Updated on'
+    required: true
+    hasDefault: false
+    writable: false
+    options: { }
+  notificationMessage:
+    type: 'Formattable'
+    name: 'Message'
+    required: false
+    hasDefault: false
+    writable: true
+    options: { }
+    location: '_meta'
+  project:
+    type: 'Project'
+    name: 'Project'
+    required: false
+    hasDefault: false
+    writable: true
+    location: '_links'
+    _links:
+      allowedValues:
+        href: '/api/v3/memberships/available_projects?filters=%5B%7B%22principal%22%3A%7B%22operator%22%3A%22%21%22%2C%22values%22%3A%5B%2214%22%5D%7D%7D%5D'
```

Comment:
I admit, that it can be confusing, but our API does only fill out `allowedValues` if there is already ANY value posted against the form endpoint. e.g. run a request `POST /api/v3/memberships/form` with the body

```json
{
  "_links": {
    "principal": {
      "href": "/api/v3/users/14"
    }
  }
}
```

and you will see those links.

Reply:
True for the form endpoints.

And this specific `membership-schema-response.yml` schema is used only to describe the `/api/v3/memberships/schema` endpoint, and on this endpoint, the returned `_links` are empty. Is it correct?

Also on the form-create and form-update endpoints, the schema are not quite the same as the project and principal are writable for create, but not for update.

Reply:
Yes, I noticed that. But this is also the reason for those links not being `required`. But if the CAN be in there, they must be specified.